integration_test/ops_agent_test.go

```diff
@@ -2091,6 +2093,386 @@ func TestPrometheusMetricsWithJSONExporter(t *testing.T) {
+			}
+		}
+
+		// 3. Config and start the agent
+		// Set the scrape interval to 10 second, so that metrics points can be
+		// received faster to shorten the duration of this test
+		config := `metrics:
+  receivers:
+    prom_app:
+      type: prometheus
+      config:
+        scrape_configs:
+        - job_name: json
+          metrics_path: /data.json
+          scrape_interval: 10s
+          static_configs:
+            - targets:
+              - localhost:8000
+  service:
+    pipelines:
+      prom_pipeline:
+        receivers: [prom_app]
+`
+		// Turn on the prometheus feature gate.
+		if err := gce.SetEnvironmentVariables(ctx, logger.ToMainLog(), vm, map[string]string{"UNSUPPORTED_BETA_PROMETHEUS_RECEIVER": "enabled"}); err != nil {
+			t.Fatal(err)
+		}
+		if err := setupOpsAgent(ctx, logger, vm, config); err != nil {
+			t.Fatal(err)
+		}
+
+		// Wait long enough for the data to percolate through the backends
+		// under normal circumstances. Based on some experiments, 2 minutes
+		// is normal; wait a bit longer to be on the safe side.
+		time.Sleep(3 * time.Minute)
+		window := time.Minute
+		var multiErr error
+
+		// 4. Wait for the initial set of metrics and check
+		// For Histogram: We use prometheus.LinearBuckets(0, 20, 5), to have
+		// buckets w/ le=[0, 20, 40, 60, 80] plus the +inf final bucket
+		// For step 1, we observe points [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
+		// And get:
+		// Bounds (less than or equal) |0  |20     |40     |60     |80     |+inf
+		// Points                      |[0]|[10,20]|[30,40]|[50,60]|[70,80]|[90]
+		// Count                       |1  |2      |2      |2      |2      |1
+		// And histogram metrics are stored as cumulative type metrics, so for
+		// this initial step, Count/Mean/SumOfSquaredDeviation are all zeros,
+		// and the BucketCounts is nil
+		stepOneExpectedHistogram := &distribution.Distribution{
+			Count:                 0,
+			Mean:                  0,
+			SumOfSquaredDeviation: 0,
+			BucketOptions: &distribution.Distribution_BucketOptions{
+				Options: &distribution.Distribution_BucketOptions_ExplicitBuckets{
+					ExplicitBuckets: &distribution.Distribution_BucketOptions_Explicit{
+						Bounds: []float64{0, 20, 40, 60, 80},
+					},
+				},
+			},
+		}
+		multiErr = multierr.Append(multiErr, assertPrometheusHistogramMetric(ctx, logger, vm, "test_histogram", window, stepOneExpectedHistogram))
+		// For Summary: We use Objectives [0.5, 0.9, 0.99]
+		// For step 1, we observe points [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
+		// And get:
+		// Objectives |0.5            |0.9               |0.99
+		// Points     |[0,10,20,30,40]|[..., 50,60,70,80]|[...,90]
+		// Quantile   |40             |80                |90
+		// And summary metrics' quantiles are stored as gauge type metrics, so
+		// for this initial step, quantiles have the actual values.
+		// But count and sum are stored as cumulative values, so those two are 0
```

Comment:
Same question for these values like for the histogram. It feels like the first points are basically just dropped?

Reply:
Summary's count and sum are cumulative and get normalized as well.